include/ghex/communication_object_2.hpp

```diff
@@ -22,6 +22,10 @@
 #include <stdio.h>
 #include <functional>
 
+// Define this macro to use optimizations for same type fields
+// Improves performance for gpu, but only when solely same field types are passed to an exhange
+//#define GHEX_COMM_OBJ_USE_U
+
```

Comment:
Line 269 takes 2 iterators (of buffer infos) as opposed to 2 (potentially different) buffer infos. That is, the member function at line 269 takes a range, whereas the function at line 259 takes several buffer infos which are not necessarily of the same type. Does this make sense?

Reply:
Ok, I got confused by the `disable_if_buffer_info<Iterator,handle_type>`, which actually is necessary because you do not want regular `buffer_info`s to fall in there, but only iterators to `buffer_info`.. Correct?

Reply:
exactly